### PR TITLE
Jinchao add seeSummaryIndicator and seeVisibilityIcon permissions

### PR DESCRIPTION
**Pull request by suigunfjc**
# Description
image

This PR creates the permissions for seeing weekly summary indicator #880 and visibility icon #859. Now owner can assign these two permissions to user or role to let them view the indicators and icons.

## Main changes explained:
- Create two permissions in `RolePermissions.jsx` and `UserRoleTab.jsx` for both role permissions and user permissions list.
- Replace boolean 'isAdmin' with `hasSummaryIndicatorPermission` and `hasVisibilityIconPermission` in `Leaderboard.jsx`

## How to test:
1. check into current branch
2. log in as owner user (if you don't have one, use my account first and create your own. jinchaoowner@gmail.com password:Ff12345678)
3. Other links->permissions management -> any role. Check if there are two new permissions at the bottom of the role permission list
image
4. Assign/unassign these permissions seperately to different roles, check if they are working as intended(don't forget to click save after you make changes):
image
image
5. Unassign permissions to all roles, then go to Other links->permissions management -> Manage user permissions, check if there are new permissions at the bottom of the permission list.
image
6. Assign permissions to one user and log in as that user to check if it works as expected.

## Note:
After changing the permission on roles, it may take more time to reload the page since it changes the `state` of the page.

### CODE REVIEW

**Review by tdkent — DISMISSED** [review]
Local testing confirms

- Selectable options `See Summary Indicator` and `See Visibility Icon` have been added as the last items of the **Manage User Permissions** form.
- Activating these options adds corresponding icon to the applicable users in the **Leaderboards** list on the Dashboard.
- Icons are visible to all roles when their permissions are active (could not confirm visibility icon for Volunteer role)

Nice work!

Testing Video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/3aafafd3-d659-460c-bdae-1aa24aa9a7a7

**Comment by luisarevalo21** [comment]
Hello [suigunfjc](https://github.com/suigunfjc)

I reviewed you PR and found the following:

- Was able to view the new permission at the bottom via my own owner account
- Was able to toggle the permissions and and assign it to each of the roles in the system
- Provided the permission to both Admin and Volunteers accounts.
- Logged into the Admin account and can view each of the icons for weekly summary submission and visibility icon.
- Logged into Volunteer account, however, couldn't retrieve the dashboard. Unsure if that is part of the volunteers permissions. 
- I also removed the permission from both the role and the account and was reverted back to normal.

Great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/db99d5e9-b7ed-4052-ace3-58c6141b77a4

Screenshot 2023-07-09 at 9 36 51 PM

**Review by AaronPersaud — DISMISSED** [review]
Good evening, I've tested it and it looks good!

<b>Before adding permissions </b>

before

<b> Adding permissions and logging back in</b>

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/10265513/e003b780-6303-4691-b7d7-d49e663d81ee

- verified two new permissions appear at the bottom of permissions list
- applying permissions shows corresponding icon on leaderboard

**Comment by freecode23** [comment]
Works smoothly. Tested for with and without those two permissions for : admin, volunteer, mentor, manager, and core team roles

**Review by BaileyM96 — DISMISSED** [review]
Good evening @suigunfjc I tested your changes and everything works fine.

## Video Walkthrough

[Manage Permissions.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108101478/221dec8a-dbe4-49b7-9e3d-08fa148fb07b)

**Comment by beblicarl** [comment]
I can confirm that this functionality works and the code looks good 

[screencast-hgnapplication_react_beta.surge.sh-2023.07.12-10_30_24.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/fe1b47d7-ede3-465d-ad62-c93dd2c2ffd2)

**Review by beblicarl — DISMISSED** [review]

**Comment by Jiangwei-shi** [comment]
two new permissions appear at the bottom of the permissions list: yes
applying permissions shows the corresponding icon on the leaderboard: yes
works as indeed. good job!

image

image

**Review by freecode23 — DISMISSED** [review]

**Review by iTvX — CHANGES_REQUESTED** [review]
The code looks good, but on my side, it doesn't work at all.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/8d85fb33-f4c2-43da-ab46-93d9c2c8fb68

I followed the instructions, created an owner account first, and then used that owner account to change the permissions of my other account. However, after switching to that account, I couldn't see any icon indicating that I have those two permissions.

Next, I switched back to the owner account and checked my test account again. I could see that the permissions had already been added to my test account.

Finally, I tried changing my browser, but the permissions icon still didn't appear.

**Comment by bienzguoa** [comment]
Good job! It works as intended! 
For owners:
Screenshot 2023-07-15 at 2 25 40 AM
Screenshot 2023-07-15 at 2 30 22 AM
delete these two permissions:
Screenshot 2023-07-15 at 2 31 00 AM
For admins:
Screenshot 2023-07-15 at 2 32 33 AM
Screenshot 2023-07-15 at 2 32 59 AM

**Review by bienzguoa — DISMISSED** [review]
Code looks clear and workable.

**Comment by one-community** [comment]
Needs branch conflicts fixed and the change request addressed and we can move this to final review.

**Comment by one-community** [comment]
> The code looks good, but on my side, it doesn't work at all.

This video is blank... just need the branch conflicts resolved and we'll move to final review.

**Comment by iTvX** [comment]
> > The code looks good, but on my side, it doesn't work at all.
> 
> This video is blank... just need the branch conflicts resolved and we'll move to final review.

On my side, the video is fine though. Could you please download it? I think it might be h265 or others issue. I will change record tool though.

**Review by iTvX — APPROVED** [review]

File: src/components/LeaderBoard/Leaderboard.jsx [hunk 1]
```diff
@@ -279,7 +282,7 @@ const LeaderBoard = ({
                     }
                   </div>
                   {
-                    isAdmin && item.hasSummary && 
+                    hasSummaryIndicatorPermission && item.hasSummary &&
```
Comment (iTvX) [inline src/components/LeaderBoard/Leaderboard.jsx:285]: the variables hasSummaryIndicatorPermission and hasVisibilityIconPermission are used multiple times. To improve code readability and maintainability, I suggest considering grouping the definitions and uses of these variables together or using more descriptive variable names.

**Comment by Lucas-E** [comment]
@suigunfjc Great job! Please resolve the conflicts so I can mot this PR to final review! Please DM me on Slack when you do it!

**Review by xaanders — DISMISSED** [review]
Hi! You've done good job. Everything works with manager, volunteer and admin users


- Manager:



- With admin user:



- And volunteer:

**Review by eduvarjaor — DISMISSED** [review]
Everything working as intended. Great job!

**Comment by Shreyj14** [comment]
the PR works as intended. I was able to see the visibility icon when given the permission

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118795427/5c55a047-8334-4282-877f-c68ecfcfee96

**Review by Shreyj14 — DISMISSED** [review]

**Review by EdwinLauM — APPROVED** [review]
Hey Jinchao! Changes work as intended! Good job! Please resolve branch conflicts as well

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/78256c9f-71f3-4c62-9c06-9c8acc78c29e

**Review by yurisokolovicz — DISMISSED** [review]
Hi @suigunfjc,

I have reviewed the PR, and it does what was intended. Using the owner account, I was able to see the two new permissions at the bottom of the "Manage User Permissions" page. Subsequently, when I logged in as an ADM account without these two permissions, their respective icons did not appear. Finally, after logging in again with the owner account and setting the "See Summary Indicator" and "See Visibility Icon" for my admin account, I was able to see these icons as an administrator.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/121908988/5afd7da7-e6c1-42d5-947e-2cbb5e2687c7

**Review by Indujaks — APPROVED** [review]
Tested changes, works as expected, can add both permissions!
Screenshot 2023-07-28 at 12 35 11 PM

**Comment by one-community** [comment]
Thank you all, will merge as soon as branch conflicts are resolved.

**Comment by suigunfjc** [comment]
@one-community conflicts resolved

**Pull request merged**